04-prompt-engineering-fundamentals/translations/ja-jp/README.md

```diff
@@ -0,0 +1,396 @@
+
+大規模言語モデル (LLM) では、プロンプトの書き方がとても重要で、慎重に作成したプロンプトは、そうでないものに比べ良い結果をもたらします。しかし、プロンプトやプロンプト・エンジニアリングとは一体どういう物なのでしょうか？また、LLM に送信する内容をどのようにして改善すればいいのでしょうか？この章と次の章では、そうした疑問に答えることを目指しています。  
+
+_生成系 AI_ は、ユーザーからの依頼に対して、テキスト、画像、オーディオ、コード等の新しいコンテンツを生み出す能力を有しています。これを実現するために、OpenAI の GPT（Generative Pre-trained Transformer）シリーズのような、自然言語とコードの使用目的でトレーニングされた _大規模言語モデル（LLM）_ を利用します。
+
+利用者は、チャットのような馴染み深い手法で、特別な技術の知識や研修を受講しなくても、これらのモデルと対話できるようになりました。これらのモデルは、_プロンプト_ で操作し、ユーザーがテキストの入力（プロンプト）を送ると、AI からの回答（コンプリーション）が得られます。その後、利用者は何度も 「AI と対話」を重ね、複数回のやり取りを通じてプロンプトを洗練させ、期待する回答が得られるまで調整を行います。
+
+現在、「プロンプト」は生成型 AI アプリケーションの主要なプログラミングインターフェースとなり、モデルに対して何を行うべきか指示し、返される回答の品質に影響を与えています。「プロンプト・エンジニアリング」は、大規模で一貫性のある高品質な回答を得るためのプロンプトの設計と最適化に焦点を当てた、急速に成長している研究分野です。
+
+## 学習目標
+
+このレッスンでは、プロンプト・エンジニアリングとは何か、そしてその重要性、また特定のモデルやアプリケーションの目的に応じた、効果的なプロンプトの作成方法について学習します。プロンプト・エンジニアリングの基本概念とベストプラクティスを理解し、実際にこれらのコンセプトを適用している例を、インタラクティブな Jupyter ノートブックの「サンドボックス」環境で動かし、その操作方法も学びます。
+
+このレッスンの終了後、下記ができるようになります：
+
+1. プロンプト・エンジニアリングとは何か、またその重要性について説明できる
+2. プロンプトの構成要素とその使用方法について述べる
+3. プロンプト・エンジニアリングのベストプラクティスと技術を習得する
+4. OpenAI のエンドポイントにアクセスし、学んだテクニックを実際に試す
+
+## サンドボックス (Sandbox) とは
+
+プロンプト・エンジニアリングは現在、科学というよりは、むしろ芸術に近いものです。これに対する感覚を高めるには、_より多くの練習_ を積み重ね、アプリケーション専門知識で推奨される技術や、モデル固有の最適化を組み合わせた、試行錯誤の方法を取り入れることが重要です。
+
+このレッスンに付随するJupyterノートブックは、学んだ内容を実際に試すことができる _サンドボックス_ 環境を提供します。レッスン中、もしくは最終的なコードチャレンジで利用することができます。演習を行うためには、下記が必要です：
+
+1. OpenAI API キー - デプロイした大規模言語モデル（LLM）のサービスエンドポイント
+2. Python の実行環境 - ノートブックを実行する環境
+
+このリポジトリは Python 3 の実行環境を含む、_.devcontainer_ を提供しています。GitHub Codespaces、もしくは Docker Desktop をインストールしたご自身のローカル環境でリポジトリを開くことで、コンテナ環境が自動的に起動します。その後、ノートブックを開いて Python 3.x カーネルを選択することで、実行するためのノートブックを準備できます。
+
+デフォルトで用意されているノートブックは、OpenAI の API キーを利用する設定になっています。フォルダのルート・ディレクトリにある`.env.copy`を`.env`にリネームし、`OPENAI_API_KEY=`の行に API キーを入力するだけで、準備が整います。
+
+このノートブックには基本的な演習が用意されていますが、より多くのサンプルやアイディアを試すために、ご自身で_Markdown_（説明文）や_Code_（プロンプトのリクエスト）のセクションを追加することができます。これにより、プロンプトデザインに対する感覚をより養うことができます。
+
+## スタートアップ
+
+それでは、_このトピック_ がスタートアップのミッションである「[教育へ AI 革新をもたらす](https://educationblog.microsoft.com/2023/06/collaborating-to-bring-ai-innovation-to-education?WT.mc_id=academic-105485-yoterada)」とどのように結びついているのかについて見て行きましょう。私たちは　_個別学習_　を実現する AI アプリケーションの開発を目指しています。そこで、私たちのアプリケーションを利用する様々な利用者がプロンプトをどのように「デザイン」するかを考えてみましょう。  
+
+- **管理者**: AI に _カリキュラムのデータを分析し、カバーしてない領域を特定する_ よう要求する可能性があります。AI はその結果をまとめたり、コードを用いて可視化することができます。
+- **教員**: AI に _特定の生徒と授業科目に応じた授業計画を作成する_ 可能性があります。AI は指定されたフォーマットに従って個別に計画を作成できます。
+- **生徒**: AI に _苦手な科目で個別指導を依頼する_ 可能性があります。AI は生徒のレベルに合わせたレッスン、ヒント、例を提供して指導することができます。
+
+これらは、ほんの一例です。教育専門家から厳選された、オープンソースのプロンプト・ライブラリ「[Prompts For Education](https://github.com/microsoft/prompts-for-edu/tree/main?WT.mc_id=academic-105485-yoterada)」を確認し、さらに広い視野で可能性を探ってみてください！ _サンドボックスでそれらのプロンプトを試しに実行したり、OpenAI Playground で試してどのような結果が出力されるか試してください！_  
+
+
+<!--
+LESSON TEMPLATE:
```

Comment:
This 'LESSON TEMPLATE:' seems like it's meant as a note for instructors. I've counted at least five instances. Do you plan not to translate this section? 🤔 

https://github.com/microsoft/generative-ai-for-beginners/pull/230/files#diff-03ee89d324536f00c2c930a03b7bd758590d83d23590c31192f561099e4166b3R50-R58

https://github.com/microsoft/generative-ai-for-beginners/pull/230/files#diff-03ee89d324536f00c2c930a03b7bd758590d83d23590c31192f561099e4166b3R169-R177

https://github.com/microsoft/generative-ai-for-beginners/pull/230/files#diff-03ee89d324536f00c2c930a03b7bd758590d83d23590c31192f561099e4166b3R295-R304

https://github.com/microsoft/generative-ai-for-beginners/pull/230/files#diff-03ee89d324536f00c2c930a03b7bd758590d83d23590c31192f561099e4166b3R337-R346

https://github.com/microsoft/generative-ai-for-beginners/pull/230/files#diff-03ee89d324536f00c2c930a03b7bd758590d83d23590c31192f561099e4166b3R372-R375